docs/metrics.md

```diff
@@ -0,0 +1,561 @@
+# OpenTelemetry Rust Metrics
+
+<details>
+<summary>Table of Contents</summary>
+
+* [Best Practices](#best-practices)
+* [Metrics API](#metrics-api)
+    * [Meter](#meter)
+    * [Instruments](#instruments)
+* [MeterProvider Management](#meterprovider-management)
+* [Memory Management](#memory-management)
+    * [Example](#example)
+    * [Pre-Aggregation](#pre-aggregation)
+    * [Cardinality Limits](#cardinality-limits)
+    * [Memory Preallocation](#memory-preallocation)
+* [Metrics Correlation](#metrics-correlation)
+* [Metrics Enrichment](#metrics-enrichment)
+
+</details>
+
+## Best Practices
+
+// TODO: Add link to the examples, once they are modified to show best
+practices.
+
+## Metrics API
+
+### Meter
+
+:stop_sign: You should avoid creating
```

Comment:
> You should avoid creating [Meter](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Meter.html) too frequently.

I think we should be more specific here. "avoid creating Meter too frequently" doesn't really capture the ask well. Maybe something like:
- "You should avoid creating duplicate `Meter` instances."
- Or "You should avoid creating `Meter` instances with the same name."